src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js

```diff
@@ -94,6 +96,10 @@ const convertInclude = async (includeBlock, ...restParams) => {
     return null;
   }
 
+  const includeMap = blocks.filter(block => block.type === 'include');
+
+  const indexOfInclude = includeMap.indexOf(includeBlock);
```

Comment:
This is ingenious but we are now interating the blocks twice, once to process them all, and again to derive the indexes. I was thinking we could maybe pass in the index of the block here: https://github.com/bbc/simorgh/blob/include-index-identifier/src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/index.js#L72 

That's less iteration but then maybe the code structure isn't great then either? 🙈

Reply:
Just had a look, that is consistent with the social embed implementation: https://github.com/bbc/simorgh/blob/include-index-identifier/src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/socialEmbed/index.js#L5 

Another alternative is using a hashing approach you can take the include url and put it into a [hashing library](https://github.com/MatthewBarker/hash-string):
`hash('/idt2/111-222-333-444-555')` -> `444207399`
You can then use that in the include id: `include-444207399`

This will give you a deterministic id for each include but prevent you from having to find out where it occurs in the block structure, it does though have a limitation in that if an include with the same url is used more than once, you would get an id collision, I'm not sure that should happen though 🤔 

If you stick with the index approach, I'd consider extracting out the logic to a utility shared by both social embeds and includes so it's clearer that it's a shared pattern used for multiple block types.

Reply:
The hashing approach seem like a really good method. I think I do prefer it more than the index approach, as unique ID's per include make the most sense. But as you've pointed out as a concern, there could be ID collisions. I've spoken with Jim and there could potentially be pages which might share the same include pulled from the same URL. Not sure how concerned we are about this, I think if multiple ID's exist then the browser will only render the first one which would still lead the user to the same include so might not be a massive deal.

There's definitely examples where journalists use the same URL ([here](https://www.bbc.com/arabic/institutional-49283069.json#formattedJson:~:text=%22href%22%3A%20%22%2Fnews%2Fspecial%2F2016%2Fnewsspec_14439%2Fanchor.inc%3Fanchor%3Dauthorexpertise%26responsive%3Dtrue%22%2C) and [here](https://www.bbc.com/arabic/institutional-49283069.json#formattedJson:~:text=%22href%22%3A%20%22%2Fnews%2Fspecial%2F2016%2Fnewsspec_14439%2Fanchor.inc%3Fanchor%3Dtypeofwork%26responsive%3Dtrue%22%2C)) with query params but they should work with the hashing approach.

I agree with the logic extraction. The current approach also has to process the blocks each time an include is rendered on the page which isn't great.

Reply:
- It's worth mentioning that the reason we're using indexes as opposed to hashes is because they're easy to calculate, guaranteed to be unique and identify Includes in an understandable way for users. Include blocks are already uniquely hashed - we're not trying to solve that problem in this PR.
- Whilst I agree with extracting shared logic, it is not the same as Social Embeds - we are doing something different there. I like the simplicity of the current implementation and would question making it more complex without good reason.

Reply:
> Include blocks are already uniquely hashed - we're not trying to solve that problem in this PR.
If the include blocks are already uniquely hashed can we not just use that, as part of the id, I haven't seen that anywhere though?
> Whilst I agree with extracting shared logic, it is not the same as Social Embeds 
Oh yeah, I've had a look it's _similar_, but not the same, yeah fair point to keep it seperate 👍 

I suggested an alternative because the include converter component is now coupled to the `pageData`, previously it only cared about the structure of an include block, now it needs to be aware of the `pageData` it is delivered within meaning this component may have to change if the `pageData` structure changed. Probably won't happen but just explaining where my unease comes from! Happy to go with the current solution.